include/pisa/topk_queue.hpp

```diff
@@ -58,26 +78,52 @@ struct topk_queue {
         m_q.resize(size);
     }
 
+    /// Returns the elements in the queue. Typically, this should be called after `finalize`
+    /// to retrieve the final results in the correct order.
     [[nodiscard]] std::vector<entry_type> const& topk() const noexcept { return m_q; }
 
-    void set_threshold(Threshold t) noexcept { m_threshold = t; }
+    /// Returns the threshold based on the heap state, defined as the score of the `k`-th document,
+    /// or 0.0 if the heap is not full.
+    Threshold true_threshold() const noexcept
+    {
+        return capacity() == size() ? m_q.front().first : 0.0;
+    }
+
+    /// Returns the threshold set at the start (by default 0.0).
+    Threshold initial_threshold() const noexcept { return m_initial_threshold; }
+
+    /// Returns the maximum of `true_threshold()` and `initial_threshold()`.
+    Threshold effective_threshold() const noexcept { return m_effective_threshold; }
```

Comment:
Again, I think this might be OK, or we might actually want to be using the max here depending on how the previous cases are handled.